src/javaRosa.js

```diff
@@ -560,10 +560,11 @@ define([
         block.refreshMessages = function () {
             // TODO improve this to display each message beside the
             // form and language to which it applies
-            if (options.messagesPath) {
-                var messages = widgets.util.getMessages(mug, options.messagesPath);
-                $messages.empty().append(messages);
-            }
+            _.each(block.languages, function(lang) {
+                var name = 'itext-' + lang + '-' + block.itextType,
+                    messages = widgets.util.getMessages(mug, name);
+                $('[name='+name+']').parent().parent().find('.messages').empty().append(messages);
```

Comment:
Would be great to do a local search rather than doing a global search on the DOM. Would `$blockUI.find('[name='+name+']')...` work?